modules/dnn/src/layers/cpu_kernels/softmax.cpp

```diff
@@ -0,0 +1,159 @@
+// of this distribution and at http://opencv.org/license.html.
+
+// This file is modified from the ficus (https://github.com/vpisarev/ficus/blob/master/lib/NN/OpNN.fx).
+// Here is the original license:
+/*
+    This file is a part of ficus language project.
+    See ficus/LICENSE for the licensing terms
+*/
+
+#include "../../precomp.hpp"
+#include "softmax.hpp"
+
+#ifdef CV_SIMD
+#define _VEXP_INIT() \
+    v_float32 _vexp_lo = vx_setall_f32(-88.3762626647949f); \
+    v_float32 _vexp_hi = vx_setall_f32(88.3762626647949f); \
+    v_float32 _vexp_half = vx_setall_f32(0.5f); \
+    v_float32 _vexp_one = vx_setall_f32(1.f); \
+    v_float32 _vexp_LOG2EF = vx_setall_f32(1.44269504088896341f); \
+    v_float32 _vexp_C1 = vx_setall_f32(-0.693359375f); \
+    v_float32 _vexp_C2 = vx_setall_f32(2.12194440e-4f); \
+    v_float32 _vexp_p0 = vx_setall_f32(1.9875691500E-4f); \
+    v_float32 _vexp_p1 = vx_setall_f32(1.3981999507E-3f); \
+    v_float32 _vexp_p2 = vx_setall_f32(8.3334519073E-3f); \
+    v_float32 _vexp_p3 = vx_setall_f32(4.1665795894E-2f); \
+    v_float32 _vexp_p4 = vx_setall_f32(1.6666665459E-1f); \
+    v_float32 _vexp_p5 = vx_setall_f32(5.0000001201E-1f)
+
+#define _VEXP_COMPUTE(x, y) { \
+    v_float32 _vexp_, _vexp_x, _vexp_y, _vexp_z; \
+    _vexp_x = v_min(x, _vexp_hi); \
+    _vexp_x = v_max(_vexp_x, _vexp_lo); \
+    _vexp_ = v_fma(_vexp_x, _vexp_LOG2EF, _vexp_half); \
+    v_int32 _vexp_mm = v_floor(_vexp_); \
+    _vexp_ = v_cvt_f32(_vexp_mm); \
+    _vexp_mm = v_add(_vexp_mm, vx_setall_s32(0x7f)); \
+    _vexp_mm = v_shl(_vexp_mm, 23); \
+    _vexp_x = v_fma(_vexp_, _vexp_C1, _vexp_x); \
+    _vexp_x = v_fma(_vexp_, _vexp_C2, _vexp_x); \
+    _vexp_z = v_mul(_vexp_x, _vexp_x); \
+    _vexp_y = v_fma(_vexp_x, _vexp_p0, _vexp_p1); \
+    _vexp_y = v_fma(_vexp_y, _vexp_x, _vexp_p2); \
+    _vexp_y = v_fma(_vexp_y, _vexp_x, _vexp_p3); \
+    _vexp_y = v_fma(_vexp_y, _vexp_x, _vexp_p4); \
+    _vexp_y = v_fma(_vexp_y, _vexp_x, _vexp_p5); \
+    _vexp_y = v_fma(_vexp_y, _vexp_z, _vexp_x); \
+    _vexp_y = v_add(_vexp_y, _vexp_one); \
+    y = v_mul(_vexp_y, v_reinterpret_as_f32(_vexp_mm)); \
+    }
+#endif
+
+namespace cv { namespace dnn {
+
+void softmax(Mat &dst, const Mat &src, int axis, int axisBias, int axisStep){
+    CV_Assert(src.type() == CV_32F);
+    CV_Assert(src.isContinuous() && dst.isContinuous());
+    CV_Assert(src.size == dst.size);
+    axis = normalize_axis(axis, src.dims);
+
+    size_t outerSize = src.total(0, axis),
+           innerSize = src.total(axis + 1);
+
+    const float *srcPtr = src.ptr<float>();
+    float *dstPtr = dst.ptr<float>();
+
+    size_t outerStep = src.total(axis);
+    size_t cnStep = src.total(axis + 1);
+
+    // multi-threads
+    size_t totalTasks = outerSize * innerSize;
+    double nstripes = (double) totalTasks / 1024.0;
+    // make the channel axis to be multiple of 8
+    size_t channelAxis = (axisStep + 7) & -8;
+
+#ifdef CV_SIMD
```

Comment:
Not sure whether definition of `CV_SIMD` guarantee availability of universal intrinsics. `#if CV_SIMD` is much more safe I think. Correct me if I am wrong.